core/src/main/kotlin/behavior/channel/threads/ThreadChannelBehavior.kt

```diff
@@ -0,0 +1,129 @@
+package dev.kord.core.behavior.channel.threads
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.common.exception.RequestException
+import dev.kord.core.Kord
+import dev.kord.core.behavior.channel.MessageChannelBehavior
+import dev.kord.core.cache.data.toData
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.thread.ThreadChannel
+import dev.kord.core.entity.channel.thread.ThreadUser
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.rest.builder.channel.thread.ThreadModifyBuilder
+import kotlinx.coroutines.flow.Flow
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+interface ThreadChannelBehavior : MessageChannelBehavior {
+    /**
+     * Requests to get all members of the current thread.
+     *
+     * The returned flow is lazily executed, any [RequestException] will be thrown on
+     * [terminal operators](https://kotlinlang.org/docs/reference/coroutines/flow.html#terminal-flow-operators) instead.
+     */
+    val members: Flow<ThreadUser>
+        get() = supplier.getThreadMembers(id)
+
+    /**
+     * Removes the user identified by [id] from the current thread.
+     * Requires the thread is not archived.
+     * Sending a message to the thread automatically unarchived it.
+     *
+     * @see [ThreadChannel.isArchived]
+     * @see [ThreadChannel.isLocked]
+     * @throws [RequestException] if something went wrong during the request.
+     */
```

Comment:
This should mention that it requires `MANAGE_THREADS` for public threads, or that the bot is the creator of the thread for private ones.